### DISCUSSION TITLE
Self-Hosted Giscus on Enterprise Account

### DISCUSSION THREAD

**Discussion by SannaBerg** [giscus/giscus#912] [Q&A]
I am looking into using Giscus to add commenting to our Hugo-built documentation site. I have a few questions that I am hoping you can help me with:

### Q1
**The hugo site repository is stored on an enterprise GitHub account and we will need to deploy the web application on Google Cloud.** Do you know if Giscus will work under these conditions? Any considerations we need to take?

### Q2
To get to know Giscus better, I have tried to set up a self-hosted version of Giscus on Vercel and add it to my regular GitHub account, but I can't get it to work. 

**What I have done:** I followed the instructions in https://github.com/giscus/giscus/blob/main/SELF-HOSTING.md. However, I skipped "Configure Supabase for caching access tokens (optional)".   So I have created a GitApp and installed it to my repository, I have set the repository to public and enabled discussions, I have copied the Giscus repository and deployed it on Vercel, entering the environment variables through their interface.  

**However, when I enter my repository in my self-hosted web app it cannot find the repository.** See linked image. **Do you have any idea why this can be?** Could this be due to some problem with the GitHub tokens and secrets? What role does the Git App and the WebApp play in Giscus functioning? What is the communication flow?
![Screenshot 2023-02-21 at 12 40 53](https://user-images.githubusercontent.com/19339407/220927777-7e8248d2-b752-4e1e-b3b2-fb3119c2f005.png)

**Comment by shayanhoshyari** [comment]
I self hosted it on AWS EKS (we have eks as we also deploy other things on it). It worked like a charm.

I put the git clone and yarn install and build in a docker container. and set yarn start as entry point.

Then made a deployment in the EKS and injected all the necessary secrets and info as env vars (and not .env.local to not put secrets in container)

I also made a github app under out enterprise

i skipped supabase caching

**Reply by shayanhoshyari** [reply]
make sure to define the ORIGIN variables otherwise all calls return 500 if the request does not set ORIGIN header, which the onboarding UI does not — and you see the behavior you are seeing

also for debugging this you don’t need to deploy to vercel, just start server on localhost
